### ISSUE TITLE
`.formatter.exs` not applied when the formatter dep is added in the same igniter call

### ISSUE THREAD

**Issue by maennchen** [ash-project/igniter#85]
**Describe the bug**
When calling the following code, all phoenix router functions will have brackets added.

**To Reproduce**

```elixir
Igniter.Test.test_project()
|> Igniter.Project.Formatter.import_dep(:phoenix)
|> Igniter.Project.Module.create_module(TestWeb.Router, """
defmodule TestWeb.Router do
  use TestWeb, :router

  pipeline :browser do
    plug :accepts, ["html"]
    plug :fetch_session
    plug :fetch_flash
    plug :protect_from_forgery
    plug :put_secure_browser_headers
  end

  scope "/", TestWeb do
    pipe_through :browser

    get "/", PageHtmlController, :home
  end
end
""")
|> Igniter.Test.puts_diff()
```

=>

```
Update: .formatter.exs

1 1   |# Used by "mix format"
2 2   |[
3   - |  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"]
  3 + |  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"],
  4 + |  import_deps: [:phoenix]
4 5   |]
5 6   |


Create: lib/test_web/router.ex

1  | defmodule TestWeb.Router do
2  |   defmodule TestWeb.Router do
3  |     use TestWeb, :router
4  | 
5  |     pipeline :browser do
6  |       plug(:accepts, ["html"])
7  |       plug(:fetch_session)
8  |       plug(:fetch_flash)
9  |       plug(:protect_from_forgery)
10 |       plug(:put_secure_browser_headers)
11 |     end
12 | 
13 |     scope "/", TestWeb do
14 |       pipe_through(:browser)
15 | 
16 |       get("/", PageHtmlController, :home)
17 |     end
18 |   end
19 | end
20 |
```

**Expected behavior**

No brackets are added

** Runtime

- Elixir version: 1.17.2
- Erlang version: 27.0.1
- OS: Ubuntu 24.04
- Igniter version: branch `extensions-and-phoenix`

**Additional context**

None

**Comment by zachdaniel** [comment]
Not sure what I changed, but trying this locally appears to work for me 👍

**Comment by maennchen** [comment]
@zachdaniel I'm still getting the same result on the branch.
You might want to check out the `test/mix/tasks/oidcc_plug.gen.controller_test.exs` test here: https://github.com/erlef/oidcc_plug/tree/igniter (`igniter` branch)

**Comment by zachdaniel** [comment]
Ah, right so @maennchen that is because the dependency itself isn't compilable (because it's not present). I'm pretty sure it would work if you had `:phoenix` as an optional test-only dependency of your package. But I'd just test it once outside of your project to confirm and if so thats good enough.

**Comment by maennchen** [comment]
@zachdaniel Running it in an actual project works. However, the test should represent reality and currently it does not.

Adding `phoenix` to the `test_project` (`Igniter.Project.Deps.add_dep(igniter, {:phoenix, "~> 1.7"})`) and adding `phoenix` to the project implementing the igniter task both don't solve the issue.

**Comment by zachdaniel** [comment]
🤔 adding phoenix wouldn't work using `add_dep`, yes. You'd have to actually write the changes to the project itself. I'm not sure why adding `phoenix` to the project implementing the igniter task doesn't solve the issue. Did you add it as an optional dep in both dev & test? In that case it should work, but in all other cases there is just nothing we can do.

**Comment by maennchen** [comment]
@zachdaniel Yes, I added it for both dev & test to oidcc_plug.

**Comment by zachdaniel** [comment]
Did you do both things at the same time? adding it to the deps of the test project and the deps of the actual project?

**Comment by maennchen** [comment]
Yes, I added both.

**Comment by zachdaniel** [comment]
Okay, should be fixed in 784819f071c3ec868f5fdc6a1edd15688e7c42f0

Mind giving it a shot?

**Comment by maennchen** [comment]
@zachdaniel Works :) Thanks!
